crates/extensions/tedge_downloader_ext/src/actor.rs

```diff
@@ -0,0 +1,93 @@
+use async_trait::async_trait;
+use download::Auth;
+use download::DownloadError;
+use download::DownloadInfo;
+use download::Downloader;
+use log::info;
+use std::path::Path;
+use std::path::PathBuf;
+use tedge_actors::Actor;
+use tedge_actors::ChannelError;
+use tedge_actors::MessageReceiver;
+use tedge_actors::RuntimeError;
+use tedge_actors::Sender;
+use tedge_actors::SimpleMessageBox;
+use tedge_utils::file::PermissionEntry;
+
+#[derive(Debug, Clone)]
+pub struct DownloadRequest {
+    pub url: String,
+    pub file_path: PathBuf,
+    pub auth: Option<Auth>,
+}
+
+impl DownloadRequest {
+    pub fn new(url: &str, file_path: &Path, auth: Option<Auth>) -> Self {
+        Self {
+            url: url.into(),
+            file_path: file_path.into(),
+            auth,
+        }
+    }
+}
+
+pub type DownloadResult = Result<PathBuf, DownloadError>;
+
+#[derive(Debug)]
+pub struct DownloadResponse {
+    pub result: DownloadResult,
+}
```

Comment:
I am in between keeping struct or not.

Pros: It will be much easier to change if we want to add new members.
Cons: Since it has now only result as a member, having a struct is redundant.

Reply:
What surprise me is that this type encapsulates a `Result` preventing to use idioms like `?` or `map_err()`.

As there is no specific `impl` for that type, I would simply remove it as useless.

Reply:
I understood your point. Changed it as you suggested below comment.